api_spec.yaml

```diff
@@ -489,7 +489,7 @@ paths:
         - OAuth2 Recipe
       operationId: renameOAuth2Scope
       description: |
```

Comment:
did you check how update cascade works? Does it work on psql and mysql as expected? Do some experiment and see if it works

Reply:
Experimented with few simple cases., works flawlessly.. Will update if any edge cases finding here.

Reply:
I was able to think of these 2 scenarios where using `update on cascade` might not be feasible:
- `Recursive Cascades`: If you have a complex data model with multiple tables that have ON UPDATE CASCADE relationships, there is a risk of recursive cascades. This occurs when updates triggers a chain reaction of cascading updates across multiple tables, potentially leading to unexpected results or performance issues.

- `Circular Dependencies`: Circular dependencies can occur when two or more tables reference each other with ON UPDATE CASCADE constraints. This can create a situation where it's impossible to update or delete any records in the tables involved without violating referential integrity.

As `scope` is referenced only in `client_scopes` table, there would n't be any issue's mentioned above. Also we have  already added index to fkey's, so this would not impact performance.